pkg/curatedpackages/packageclient.go

```diff
@@ -84,14 +87,39 @@ func (pc *PackageClient) WritePackagesToStdOut(packages []packagesv1.Package) er
 	return nil
 }
 
-func (pc *PackageClient) getPackageNameToPackage() map[string]packagesv1.BundlePackage {
+func (pc *PackageClient) GetPackageFromBundle(packageName string) (*packagesv1.BundlePackage, error) {
+	packageMap := pc.packageMap()
+	p := packageMap[strings.ToLower(packageName)]
+	if p.Name != "" {
+		return &p, nil
+	}
+	return nil, fmt.Errorf("package %s not found", packageName)
+}
+
+func (pc *PackageClient) packageMap() map[string]packagesv1.BundlePackage {
 	pMap := make(map[string]packagesv1.BundlePackage)
 	for _, p := range pc.bundle.Spec.Packages {
 		pMap[strings.ToLower(p.Name)] = p
 	}
 	return pMap
 }
 
+func (pc *PackageClient) InstallPackage(ctx context.Context, bp *packagesv1.BundlePackage, customName string, kubeConfig string) error {
+	p := convertBundlePackageToPackage(*bp, customName, pc.bundle.APIVersion)
+	displayPackage := NewDisplayablePackage(&p)
+	params := []string{"create", "-f", "-", "--kubeconfig", kubeConfig}
+	packageYaml, err := yaml.Marshal(displayPackage)
+	if err != nil {
+		return err
+	}
+	stdOut, err := pc.kubectl.ExecuteCommandFromBytes(ctx, packageYaml, params...)
```

Comment:
I would maybe use `CreateFromYaml` and pass packageYaml and kubeconfig as args

Also, this will fail if the package already exists, is that the intended behavior?

Reply:
Yes if a package already exists, it should fail.